text/0457-nested-lookups.md

```diff
@@ -0,0 +1,150 @@
+
+When using angle-bracket syntax, the equivalent invocation looks like this:
+
+```hbs
+<SomeComponent @some={{parameters}}>
+```
+
+The syntax means the same thing, with one additional step:
+
+1. **normalize `SomeComponent` into `some-component` using the dasherization rule specified in [RFC #311][angle-bracket-dasherize]**
+2. look up `component:some-component` in the container
+3. look up `template:components/some-component` in the container
+4. invoke the component with its template (using the relevant component manager semantics)
+
+[angle-bracket-dasherize]: https://emberjs.github.io/rfcs/0311-angle-bracket-invocation.html#tag-name
+
+Curly bracket syntax also allows the user to invoke a component nested inside of a directory. This syntax:
+
+```hbs
+{{app-icons/warning color="yellow"}}
+```
+
+has these semantics:
+
+1. look up `component:app-icons/warning` in the container
+2. look up `template:components/app-icons/warning` in the container
+3. invoke the component with its template (using the relevant component manager semantics)
+
+However, RFC #311 didn't specify an angle-bracket equivalent syntax for the same semantics.
+
+This RFC proposes that the `::` separator serve the same purpose, with the same semantics, in angle-bracket notation:
+
+```hbs
+<AppIcons::Warning @color="yellow" />
+```
+
+---
+
+This RFC is an extension to the **normalization** rules that already occur in angle bracket notation.
+
+RFC #311 specified the normalization rules as:
+
+> The tag name will be normalized using the dasherize function, which is the same rules used by existing use cases, such as service injections
+
+This RFC amends the normalization rule by first replacing any occurrences of `::` in the tag name with a `/`, but otherwise doesn't change the semantics of RFC #311.
+
+[RFC #143][module-unification], Module Unification, uses `::` as a package separator, but because of problems with scoped packages in npm, we no longer intend to use the `::` syntax in that way. Instead, we intend to allow templates to import components from other packages using import syntax, and the `::` syntax is therefore available for this purpose.
+
+[module-unification]: https://emberjs.github.io/rfcs/0143-module-unification.html
+
+## How we teach this
+
+We currently don't cover directory nesting in the guides, and directory nesting is not an extremely early-stage Ember feature. That said, most Ember developers encounter a need or desire to group components together relatively quickly, and it does make sense to cover it.
+
+RFC #311 introduced a normalization rule for angle bracket invocation, and the guides mention that `<NavBar>` invokes a component that appears in the file system as `nav-bar`.
+
+![rental-listing in the docs](../images/457-dasherization.jpg)
+
+After this RFC, the documentation should add a "Zoey says" sidebar that describes the rule in more detail, and mentions that you can refer to components nested in a directory with the `::` separator.
```

Comment:
```suggestion
As part of shipping the feature in Ember.js, updates should be made to the API documentation for [components](https://emberjs.com/api/ember/release/classes/Component). Then, the Guides section on components should add a "Zoey says" callout that describes Angle Bracket normalization rules. The callout can mention that components are nested in a directory with the `::` separator, and it will have a link to more details in the API docs.
```